docs/rules/no-try-expect.md

```diff
@@ -0,0 +1,42 @@
+# Prevent catch assertions in tests (no-try-expect)
+
+This rule prevents the use of `expect` inside `catch` blocks.
+
+## Rule Details
+
+Expectations inside a `catch` block can be silently skipped. While Jest provides
+an `expect.assertions(number)` helper, Shopify rarely uses this feature. Using
+`toThrow` concisely guarantees that an exception was thrown, and that its
+contents match expectations.
+
+The following patterns are warnings:
+
+```js
```

Comment:
can you add an example where you test `error.code === 'MODULE_NOT_FOUND'` or something? You're not always testing the message of the error.

Should be `expect(() => foo()).toThrow(expect.objectContaining({ code: 'MODULE_NOT_FOUND' }));` in the "not warning" section